docs/ndcube.rst

```diff
@@ -0,0 +1,355 @@
+======
+NDCube
+======
+
+`~ndcube.NDCube` is the fundamental class of the ndcube package and is designed
+to handle array contained in a single N-D array described by a single
+WCS transformation.  `~ndcube.NDCube` is subclassed from `astropy.nddata.NDData`
+and so inherits the same attributes for data, wcs, uncertainty, mask,
+meta, and unit.  The WCS object contained in the .wcs attribute is
+subclassed from `astropy.wcs.WCS` and contains a few additional
+attributes to enable to keep track of its relationship to the data.
+
+Initialization
+--------------
+
+To initialize the most basic `~ndcube.NDCube` object, all you need is a
+`numpy.ndarray` containing the data, and an `astropy.wcs.WCS` object
+describing the transformation from array-element space to real world
+coordinates.  Let's create a 3-D array of data with shape (3, 4, 5)
+where every value is 1::
+
+  >>> import numpy as np
+  >>> data = np.ones((3, 4, 5))
+
+Now let's create an `astropy.wcs.WCS` object describing the
+translation from the array element coordinates to real world
+coordinates.  Let's the first data axis be helioprojected longitude, the
+second by helioprojected latitude, and the third be wavelength.  (Note
+that due to (confusing) convention, the order of the axes in the WCS
+object is reversed relative to the data array.
+
+  >>> import astropy.wcs
+  >>> wcs_input_dict = {
+  ... 'CTYPE1': 'WAVE    ', 'CUNIT1': 'Angstrom', 'CDELT1': 0.2, 'CRPIX1': 0, 'CRVAL1': 10, 'NAXIS1': 5,
+  ... 'CTYPE2': 'HPLT-TAN', 'CUNIT2': 'deg', 'CDELT2': 0.5, 'CRPIX2': 2, 'CRVAL2': 0.5, 'NAXIS2': 4,
+  ... 'CTYPE3': 'HPLN-TAN', 'CUNIT3': 'deg', 'CDELT3': 0.4, 'CRPIX3': 2, 'CRVAL3': 1, 'NAXIS3': 3}
+  >>> input_wcs = astropy.wcs.WCS(wcs_input_dict)
+
+Now that we have a data array and a corresponding WCS object, we can
+create an `~ndcube.NDCube` instance by doing::
+
+  >>> import ndcube
+  >>> my_cube = ndcube.NDCube(data, input_wcs)
+
+The data array is stored in the ``mycube.data`` attribute while the
+WCS object is stored in the ``my_cube.wcs`` attribute.  However, when
+manipulating/slicing the data is it better to slice the object as a
+whole.  (See section on Slicing.)  So the ``.data`` attribute should only
```

Comment:
link to the section on slicing?

Reply:
Good suggestion.  How do I do that?

Reply:
http://www.sphinx-doc.org/en/stable/markup/inline.html#cross-referencing-arbitrary-locations